addon/components/polaris-data-table.js

```diff
@@ -0,0 +1,535 @@
+    if (isBlank(this.get('footerContent'))) {
+      return null;
+    }
+
+    return htmlSafe(`margin-bottom: ${ this.get('heights.lastObject') }px;`);
+  }).readOnly(),
+
+  resetScrollPosition() {
+    let { left, top } = this.get('preservedScrollPosition');
+
+    if (left) {
+      this.get('scrollContainer').scrollLeft = left;
+    }
+
+    if (top) {
+      window.scrollTo(0, top);
+    }
+  },
+
+  setHeightsAndScrollPosition() {
+    this.set('heights', this.tallestCellHeights());
+
+    scheduleOnce('afterRender', this, this.resetScrollPosition);
+  },
+
+  calculateColumnVisibilityData(collapsed) {
+    if (collapsed) {
+      let headerCells = this.get('table').querySelectorAll(
+        '[class*=header]',
+      );
+      let collapsedHeaderCells = Array.from(headerCells).slice(2);
+      let fixedColumnWidth = headerCells[0].offsetWidth;
+      let tableData = {
+        fixedColumnWidth,
+        firstVisibleColumnIndex: collapsedHeaderCells.length - 1,
+        tableLeftVisibleEdge: this.get('scrollContainer').scrollLeft,
+        tableRightVisibleEdge:
+          this.get('scrollContainer').scrollLeft +
+          (this.get('dataTable').offsetWidth - fixedColumnWidth),
+      };
+      let columnVisibilityData = collapsedHeaderCells.map(
+        measureColumn(tableData),
+      );
+
+      return assign(
+        {
+          columnVisibilityData,
+        },
+        getPrevAndCurrentColumns(tableData, columnVisibilityData)
+      );
+    }
+
+    return {
+      columnVisibilityData: [],
+      previousColumn: undefined,
+      currentColumn: undefined,
+    };
+  },
+
+  handleResize() {
+    // This is needed to replicate the React implementation's `@debounce` decorator.
+    debounce(this, 'debouncedHandleResize', 0);
+  },
+
+  debouncedHandleResize() {
+    if (this.get('isDestroying') || this.get('isDestroyed')) {
+      return;
+    }
+
+    let { footerContent, truncate } = this.getProperties('footerContent', 'truncate');
+    let collapsed = this.get('table.scrollWidth') > this.get('dataTable.offsetWidth');
+
+    this.get('scrollContainer').scrollLeft = 0;
+    this.setProperties(assign(
+      {
+        collapsed,
+        heights: [],
+      },
+      this.calculateColumnVisibilityData(collapsed),
+    ));
+
+    scheduleOnce('afterRender', () => {
+      if (footerContent || !truncate) {
+        this.setHeightsAndScrollPosition();
+      }
+    });
+  },
+
+  scrollListener() {
+    if (this.get('isDestroying') || this.get('isDestroyed')) {
+      return;
+    }
+
+    this.setProperties(this.calculateColumnVisibilityData(this.get('collapsed')));
+  },
+
+  tallestCellHeights() {
+    let { footerContent, truncate, heights } = this.getProperties('footerContent', 'truncate', 'heights');
+    let rows = Array.from(this.get('table').getElementsByTagName('tr'));
+
+    if (!truncate) {
+      return (heights = rows.map((row) => {
+        let fixedCell = row.children[0];
```

Comment:
react uses `childNodes`, any reason we don't follow?

Reply:
Yep, `childNodes` includes a bunch of extra `text` nodes in Ember land which aren't present in the React world (spent a lil while tracking down why this wasn't behaving properly and that turned out to be the cause) 🤷‍♂️